rust/src/csvreader.rs

```diff
@@ -0,0 +1,263 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Can we have a separate mod for csv, i.e., `rust/src/csv` and `rust/src/csv/reader.rs`?

Reply:
done